includes/delete-comments-form.php

```diff
@@ -0,0 +1,122 @@
+		if ( $wpdb->query( "TRUNCATE $wpdb->commentmeta" ) != FALSE ){
+            if ( $wpdb->query( "TRUNCATE $wpdb->comments" ) != FALSE ){
+                $wpdb->query( "UPDATE $wpdb->posts SET comment_count = 0 WHERE post_author != 0" );
+                $wpdb->query( "OPTIMIZE TABLE $wpdb->commentmeta" );
+                $wpdb->query( "OPTIMIZE TABLE $wpdb->comments" );
+            	echo "<p style='color:green'><strong>" . __( 'All comments have been deleted.', 'disable-comments' ) . "</strong></p>";
+            } else {
+	            echo "<p style='color:red'><strong>" . __( 'Internal error occured. Please try again later.', 'disable-comments' ) . "</strong></p>";
+            }
+        } else {
+            echo "<p style='color:red'><strong>" . __( 'Internal error occured. Please try again later.', 'disable-comments' ) . "</strong></p>";
+        }
+	} else {
+		$delete_post_types =  empty( $_POST['delete_types'] ) ? array() : (array) $_POST['delete_types'];
+		$delete_post_types = array_intersect( $delete_post_types, array_keys( $types ) );
+
+		// Extra custom post types
+		if( $this->networkactive && !empty( $_POST['delete_extra_post_types'] ) ) {
+			$delete_extra_post_types = array_filter( array_map( 'sanitize_key', explode( ',', $_POST['delete_extra_post_types'] ) ) );
+			$delete_extra_post_types = array_diff( $delete_extra_post_types, array_keys( $types ) );	// Make sure we don't double up builtins
+			$delete_post_types = array_merge($delete_post_types, $delete_extra_post_types);
+		}
+
+		if ( ! empty( $delete_post_types ) ) {
+			// Loop through post_types and remove comments/meta and set posts comment_count to 0
+			foreach ( $delete_post_types as $delete_post_type ) {
+                $wpdb->query( "DELETE cmeta FROM $wpdb->commentmeta cmeta INNER JOIN $wpdb->comments comments ON cmeta.comment_id=comments.comment_ID INNER JOIN $wpdb->posts posts ON comments.comment_post_ID=posts.ID WHERE posts.post_type = '$delete_post_type'" );
+                $wpdb->query( "DELETE comments FROM $wpdb->comments comments INNER JOIN $wpdb->posts posts ON comments.comment_post_ID=posts.ID WHERE posts.post_type = '$delete_post_type'" );
+                $wpdb->query( "UPDATE $wpdb->posts SET comment_count = 0 WHERE post_author != 0 AND post_type = '$delete_post_type'" );
+				echo "<p style='color:green'><strong>" . sprintf( __( 'All comments have been deleted for %ss.', 'disable-comments' ), $delete_post_type ) . "</strong></p>";
+			}
+
+            $wpdb->query( "OPTIMIZE TABLE $wpdb->commentmeta" );
+            $wpdb->query( "OPTIMIZE TABLE $wpdb->comments" );
+
+            echo "<h4 style='color:green'><strong>" . __( 'Comment Deletion Complete', 'disable-comments') . "</strong></h4>";
+		}
+
+	}
+
+	$comments_count = $wpdb->get_var("SELECT count(comment_id) from $wpdb->comments");
+	if ( $comments_count <= 0 ) { ?>
+		<p><strong>No comments available for deletion.</strong></p>
+		</div>
+	<?php
+		exit;
```

Comment:
Should not be calling `exit;` here - there is still a load of WP core code to run after this function completes. You probably meant `return;`, but see my comment above.